File: Calendar/Calendar_Server/src/main/java/com/example/calendarserver/eventdata/Countdown.java
package com.example.calendarserver.eventdata;

import java.util.Date;

import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.IdClass;

@Entity
@IdClass(CompositeID.class)
public class Countdown {
	@Id
	private long id;
	@Id
	private long userID;
	private String title;
	private Date time;
	private Repetition repetition;
	private Reminder reminder;

	public Countdown() {
	}

	public String getTitle() {
		return title;
	}

	public void setTitle(String title) {
		this.title = title;
	}

	public Date getTime() {
		return time;
	}

	public void setTime(Date time) {
		this.time = time;
	}

	public Repetition getRepetition() {
		return repetition;
	}

	public void setRepetition(Repetition repetition) {
		this.repetition = repetition;
	}

	public Reminder getReminder() {
		return reminder;
	}

	public void setReminder(Reminder reminder) {
		this.reminder = reminder;
	}

	public long getUserID() {
		return userID;
	}

	public void setUserID(long userID) {
		this.userID = userID;
	}

	public long getId() {
		return id;
	}

	public void setId(long id) {
		this.id = id;
	}
}

File: Calendar/CalendarApp/app/src/main/java/com/example/calendar_app/data/Countdown.kt
package com.example.calendar_app.data

import android.arch.persistence.room.ColumnInfo
import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey
import android.arch.persistence.room.TypeConverters
import android.os.Parcelable
import kotlinx.android.parcel.Parcelize
import java.util.*

@Entity(tableName = "countdown")
@TypeConverters(Converters::class)
@Parcelize
data class Countdown(
    @ColumnInfo(name = "userID") var userID: Long,
    @ColumnInfo(name = "title") var title: String,
    @ColumnInfo(name = "time") var time: Date,
    @ColumnInfo(name = "reminder") var reminder: Reminder,
    @ColumnInfo(name = "repetition") var repetition: Repetition,
    @ColumnInfo(name = "id")
    @PrimaryKey(autoGenerate = true) var id: Long = 0
) : Parcelable
